crates/cli/src/subcommands/generate/mod.rs

```diff
@@ -107,20 +106,15 @@ pub fn exec(args: &clap::ArgMatches) -> anyhow::Result<()> {
     let delete_files = args.get_flag("delete_files");
     let force = args.get_flag("force");
 
-    let wasm_file = match wasm_file {
-        Some(x) => x,
-        None => match crate::tasks::build(project_path, skip_clippy, build_debug) {
-            Ok(wasm_file) => wasm_file,
-            Err(e) => {
-                return Err(anyhow::anyhow!(
-                    "{:?}
-
-Failed to compile module {:?}. See cargo errors above for more details.",
-                    e,
-                    project_path,
-                ));
-            }
-        },
+    let wasm_file = if !project_path.is_dir() && project_path.extension().map_or(false, |ext| ext == "wasm") {
+        println!("Note: Using --project-path to provide a wasm file is deprecated, and will be");
+        println!("removed in a future release. Please use --wasm-file instead.");
+        project_path.clone()
+    } else if let Some(path) = wasm_file {
+        println!("Skipping build. Instead we are inspecting {}", path.display());
+        path.clone()
+    } else {
+        crate::tasks::build(project_path, skip_clippy, build_debug)?
```

Comment:
this structure, param handling, and error behavior is more in line with what `publish` does (see below)